java/test_flatbuf1/Test.java

```diff
@@ -48,32 +279,35 @@ void run() {
         builder.insert_HI(new BigInteger("0ABACABA0ABACABA", 16));
         builder.insert_II((short)7);
         {
-            Tuple3__bool__bit_8___string ji = builder.create_Tuple3__bool__bit_8___string(true, (byte)10, "string");
+            Tuple3__bool__bit_8___stringWriter ji = builder.create_Tuple3__bool__bit_8___string(true, (byte)10, "string");
             builder.insert_JI(ji);
         }
         {
-            Tuple3__bool__bit_8___string ki = builder.create_Tuple3__bool__bit_8___string(false, (byte)9, "text");
+            Tuple3__bool__bit_8___stringWriter ki = builder.create_Tuple3__bool__bit_8___string(false, (byte)9, "text");
             builder.insert_KI(ki);
         }
         builder.insert_LI(true, (byte)-1, "something");
         builder.insert_L0I(false, (byte)-2, "else");
         {
-            boolean[] v = { true, false, true };
-            builder.insert_MI(v);
+            Boolean[] v = new Boolean[] {true, false, true};
+            builder.insert_MI(Arrays.asList(v));
         }
         {
-            Tuple3__bool__bit_8___string[] vec = {
-                builder.create_Tuple3__bool__bit_8___string(true, (byte)-1, "check"),
-                builder.create_Tuple3__bool__bit_8___string(false, (byte)1, "fails")
-            };
+            ArrayList<Tuple3__bool__bit_8___stringWriter> vec = new ArrayList<Tuple3__bool__bit_8___stringWriter>(2);
```

Comment:
This API is more wasteful: it grows the ArrayList dynamically.
I think you should probably support both List and [].

Reply:
`ArrayList` can be pre-allocated, which is what I do in the above code. I had to switch from arrays to `List`'s because I did not find a function type in Java that returns an array (I need that internally to construct a closure that returns a vector).